wazo_router_confd/services/carrier.py

```diff
@@ -3,23 +3,40 @@
 
 from sqlalchemy.orm import Session
 
+from wazo_router_confd.auth import Principal
 from wazo_router_confd.models.carrier import Carrier
 from wazo_router_confd.schemas import carrier as schema
+from wazo_router_confd.services import tenant as tenant_service
 
 
-def get_carrier(db: Session, carrier_id: int) -> Carrier:
-    return db.query(Carrier).filter(Carrier.id == carrier_id).first()
+def get_carrier(db: Session, principal: Principal, carrier_id: int) -> Carrier:
+    db_carrier = db.query(Carrier).filter(Carrier.id == carrier_id)
+    if principal is not None and principal.tenant_uuid:
+        db_carrier = db_carrier.filter(Carrier.tenant_uuid == principal.tenant_uuid)
```

Comment:
On a GET, DELETE and PUT on `/resource/<uuid>` we usually return all visible resources starting at the tenant_uuid of the token OR the specified `Wazo-Tenant`.

So for the following tenants
```
      A
     / \
   B    C
  /
D
```
Given a token with tenant_uuid = B and no Wazo-Tenant the visible tenants will be B and D
Given a token with tenant_uuid = B and Wazo-Tenant = D the visible tenants will be D
Given a token with tenant_uuid = B and Wazo-Tenant = C 401 is returned

To be clear, you do not have to "compute" the visible tenant list. The result you get from `/tenants` in the middleware already does that filtering.

So the filter here should be `Carrier.tenant_uuid.in_(principal.tenant_uuids)`
and if the `tenant_uuids` list is empty you can either return early from the function since nothing is visible or filter on `text("false")` to avoid the sqlalchemy warning on `IN` on empty lists